### PR TITLE
feat(Box): Migration + Simplification [GM-202]

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/23969

Dependencies:
- [x]   https://github.com/Codecademy/client-modules/tree/ar-dialog-migration
- [x] - https://github.com/Codecademy/client-modules/pull/1674
- [x] - https://github.com/Codecademy/client-modules/pull/1659

* Migrates Box System Props to Variance
* Sets `FlexBox` to `display: flex` and `GridBox` to `display: grid` statically to allow for users to add their own default props safely when extending these components.
* Updates all prop references and extensions to new API references `paddingX` => `px` etc.
* Updates some usages to `border={1}` shorthand as all properties are no longer needed.
* Refactors some presentational structure in a few components (e.g. [BottomTag](https://github.com/Codecademy/client-modules/pull/1658/files#diff-0a88d99d132d11cc2e6fbc3af0befc3973e56db2e35842fe02a35c92d58b9dfd)) to better rely on primitive separation of concerns.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: GM-202
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.3-alpha.88358c.0<br/>@codecademy/gamut-kit@0.3.39-alpha.88358c.0<br/>@codecademy/gamut-labs@14.0.1-alpha.88358c.0<br/>@codecademy/gamut-styles@9.5.4-alpha.88358c.0<br/>@codecademy/gamut-tests@2.3.15-alpha.88358c.0<br/>@codecademy/gamut@32.0.1-alpha.88358c.0<br/>@codecademy/konamimojisplosion@0.2.15-alpha.88358c.0<br/>@codecademy/markdown-overrides@0.5.27-alpha.88358c.0<br/>@codecademy/styleguide@36.0.1-alpha.88358c.0<br/>@codecademy/variance@0.9.1-alpha.88358c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608029a82a205822c583e828--gamut-preview.netlify.app>https://608029a82a205822c583e828--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608029a82a205822c583e828>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.3-alpha.541098.0<br/>@codecademy/gamut-kit@0.3.39-alpha.541098.0<br/>@codecademy/gamut-labs@14.0.1-alpha.541098.0<br/>@codecademy/gamut-styles@9.5.4-alpha.541098.0<br/>@codecademy/gamut-tests@2.3.15-alpha.541098.0<br/>@codecademy/gamut@32.0.1-alpha.541098.0<br/>@codecademy/konamimojisplosion@0.2.15-alpha.541098.0<br/>@codecademy/markdown-overrides@0.5.27-alpha.541098.0<br/>@codecademy/styleguide@36.0.1-alpha.541098.0<br/>@codecademy/variance@0.9.1-alpha.541098.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.3-alpha.736c05.0<br/>@codecademy/gamut-kit@0.3.39-alpha.736c05.0<br/>@codecademy/gamut-labs@14.0.1-alpha.736c05.0<br/>@codecademy/gamut-styles@9.5.4-alpha.736c05.0<br/>@codecademy/gamut-tests@2.3.15-alpha.736c05.0<br/>@codecademy/gamut@32.0.1-alpha.736c05.0<br/>@codecademy/konamimojisplosion@0.2.15-alpha.736c05.0<br/>@codecademy/markdown-overrides@0.5.27-alpha.736c05.0<br/>@codecademy/styleguide@36.0.1-alpha.736c05.0<br/>@codecademy/variance@0.9.1-alpha.736c05.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60802d9411f9a71100bf14cb--gamut-preview.netlify.app>https://60802d9411f9a71100bf14cb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60802d9411f9a71100bf14cb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.5-alpha.f982c1.0<br/>@codecademy/gamut-kit@0.3.45-alpha.f982c1.0<br/>@codecademy/gamut-labs@14.2.5-alpha.f982c1.0<br/>@codecademy/gamut-styles@9.5.6-alpha.f982c1.0<br/>@codecademy/gamut-tests@2.3.17-alpha.f982c1.0<br/>@codecademy/gamut@32.1.4-alpha.f982c1.0<br/>@codecademy/konamimojisplosion@0.2.17-alpha.f982c1.0<br/>@codecademy/markdown-overrides@0.5.29-alpha.f982c1.0<br/>@codecademy/styleguide@36.2.6-alpha.f982c1.0<br/>@codecademy/variance@0.9.1-alpha.f982c1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6083745f8ca2333513c3efd3--gamut-preview.netlify.app>https://6083745f8ca2333513c3efd3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6083745f8ca2333513c3efd3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.dc0922.0<br/>@codecademy/gamut-kit@0.3.65-alpha.dc0922.0<br/>@codecademy/gamut-labs@15.2.5-alpha.dc0922.0<br/>@codecademy/gamut-styles@9.7.1-alpha.dc0922.0<br/>@codecademy/gamut-tests@2.3.23-alpha.dc0922.0<br/>@codecademy/gamut@32.7.2-alpha.dc0922.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.dc0922.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.dc0922.0<br/>@codecademy/styleguide@36.9.2-alpha.dc0922.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b4f6f35386f189e74afc5--gamut-preview.netlify.app>https://608b4f6f35386f189e74afc5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b4f6f35386f189e74afc5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.ff2510.0<br/>@codecademy/gamut-kit@0.3.65-alpha.ff2510.0<br/>@codecademy/gamut-labs@15.2.5-alpha.ff2510.0<br/>@codecademy/gamut-styles@9.7.1-alpha.ff2510.0<br/>@codecademy/gamut-tests@2.3.23-alpha.ff2510.0<br/>@codecademy/gamut@32.7.2-alpha.ff2510.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.ff2510.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.ff2510.0<br/>@codecademy/styleguide@36.9.2-alpha.ff2510.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b57d4d1216a00b721f3cf--gamut-preview.netlify.app>https://608b57d4d1216a00b721f3cf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b57d4d1216a00b721f3cf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b590789eb3a00b755c190--gamut-preview.netlify.app>https://608b590789eb3a00b755c190--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b590789eb3a00b755c190>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.5f5fbb.0<br/>@codecademy/gamut-kit@0.3.65-alpha.5f5fbb.0<br/>@codecademy/gamut-labs@15.2.5-alpha.5f5fbb.0<br/>@codecademy/gamut-styles@9.7.1-alpha.5f5fbb.0<br/>@codecademy/gamut-tests@2.3.23-alpha.5f5fbb.0<br/>@codecademy/gamut@32.7.2-alpha.5f5fbb.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.5f5fbb.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.5f5fbb.0<br/>@codecademy/styleguide@36.9.2-alpha.5f5fbb.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.1d7cce.0<br/>@codecademy/gamut-kit@0.3.65-alpha.1d7cce.0<br/>@codecademy/gamut-labs@15.2.5-alpha.1d7cce.0<br/>@codecademy/gamut-styles@9.7.1-alpha.1d7cce.0<br/>@codecademy/gamut-tests@2.3.23-alpha.1d7cce.0<br/>@codecademy/gamut@32.7.2-alpha.1d7cce.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.1d7cce.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.1d7cce.0<br/>@codecademy/styleguide@36.9.2-alpha.1d7cce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b5ff1c334c8009a8e487e--gamut-preview.netlify.app>https://608b5ff1c334c8009a8e487e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b5ff1c334c8009a8e487e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.424cc5.0<br/>@codecademy/gamut-kit@0.3.65-alpha.424cc5.0<br/>@codecademy/gamut-labs@15.2.5-alpha.424cc5.0<br/>@codecademy/gamut-styles@9.7.1-alpha.424cc5.0<br/>@codecademy/gamut-tests@2.3.23-alpha.424cc5.0<br/>@codecademy/gamut@32.7.2-alpha.424cc5.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.424cc5.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.424cc5.0<br/>@codecademy/styleguide@36.9.2-alpha.424cc5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b652f34474414a430f19a--gamut-preview.netlify.app>https://608b652f34474414a430f19a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b652f34474414a430f19a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.11-alpha.96d6cd.0<br/>@codecademy/gamut-kit@0.3.65-alpha.96d6cd.0<br/>@codecademy/gamut-labs@15.2.5-alpha.96d6cd.0<br/>@codecademy/gamut-styles@9.7.1-alpha.96d6cd.0<br/>@codecademy/gamut-tests@2.3.23-alpha.96d6cd.0<br/>@codecademy/gamut@32.7.2-alpha.96d6cd.0<br/>@codecademy/konamimojisplosion@0.2.23-alpha.96d6cd.0<br/>@codecademy/markdown-overrides@0.5.35-alpha.96d6cd.0<br/>@codecademy/styleguide@36.9.2-alpha.96d6cd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b74bce5120819f2034a4f--gamut-preview.netlify.app>https://608b74bce5120819f2034a4f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b74bce5120819f2034a4f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.2-alpha.b3c5fc.0<br/>@codecademy/gamut-kit@0.3.67-alpha.b3c5fc.0<br/>@codecademy/gamut-labs@15.2.7-alpha.b3c5fc.0<br/>@codecademy/gamut-styles@9.7.2-alpha.b3c5fc.0<br/>@codecademy/gamut-tests@2.3.24-alpha.b3c5fc.0<br/>@codecademy/gamut@32.7.4-alpha.b3c5fc.0<br/>@codecademy/konamimojisplosion@0.2.24-alpha.b3c5fc.0<br/>@codecademy/markdown-overrides@0.5.36-alpha.b3c5fc.0<br/>@codecademy/styleguide@36.9.4-alpha.b3c5fc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609019fb2cdf17112722261f--gamut-preview.netlify.app>https://609019fb2cdf17112722261f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609019fb2cdf17112722261f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.2-alpha.d03caf.0<br/>@codecademy/gamut-kit@0.3.67-alpha.d03caf.0<br/>@codecademy/gamut-labs@15.2.7-alpha.d03caf.0<br/>@codecademy/gamut-styles@9.7.2-alpha.d03caf.0<br/>@codecademy/gamut-tests@2.3.24-alpha.d03caf.0<br/>@codecademy/gamut@32.7.4-alpha.d03caf.0<br/>@codecademy/konamimojisplosion@0.2.24-alpha.d03caf.0<br/>@codecademy/markdown-overrides@0.5.36-alpha.d03caf.0<br/>@codecademy/styleguide@36.9.4-alpha.d03caf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6090288c41f2b200e77b0887--gamut-preview.netlify.app>https://6090288c41f2b200e77b0887--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6090288c41f2b200e77b0887>Deploy Logs</a></p>

File: packages/styleguide/stories/Molecules/Coachmark/Coachmark.examples.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ export const CoachmarkExample = (args: CoachmarkProps) => {
   );
 
   const renderPopover = () => (
-    <FlexBox flexDirection="column" padding={16} alignItems="flex-start">
+    <FlexBox flexDirection="column" p={16} alignItems="flex-start">
       <Text marginBottom={8}>You should click the button.</Text>
       <FillButton
         onClick={() => {
```
Comment (dreamwasp) [inline packages/styleguide/stories/Molecules/Coachmark/Coachmark.examples.tsx:14]: this page is breaking for me, but is also happening on main

Comment (dreamwasp) [inline packages/styleguide/stories/Molecules/Coachmark/Coachmark.examples.tsx:14]: this page is breaking for me, but this also happening on main

**Review by dreamwasp — APPROVED** [review]
other than weird thing with the coachmark page, this looks ✨

File: packages/styleguide/stories/Molecules/Coachmark/Coachmark.examples.tsx (hunk 1 above)
Comment (codecaaron) [reply packages/styleguide/stories/Molecules/Coachmark/Coachmark.examples.tsx:14]: Ah yeah I remember why one sec

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.11.2-alpha.f21bb2.0<br/>@codecademy/gamut-kit@0.3.67-alpha.f21bb2.0<br/>@codecademy/gamut-labs@15.2.7-alpha.f21bb2.0<br/>@codecademy/gamut-styles@9.7.2-alpha.f21bb2.0<br/>@codecademy/gamut-tests@2.3.24-alpha.f21bb2.0<br/>@codecademy/gamut@32.7.4-alpha.f21bb2.0<br/>@codecademy/konamimojisplosion@0.2.24-alpha.f21bb2.0<br/>@codecademy/markdown-overrides@0.5.36-alpha.f21bb2.0<br/>@codecademy/styleguide@36.9.4-alpha.f21bb2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609056c0eac0c400d9eec708--gamut-preview.netlify.app>https://609056c0eac0c400d9eec708--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609056c0eac0c400d9eec708>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.13.2-alpha.39e0b4.0<br/>@codecademy/gamut-kit@0.3.70-alpha.39e0b4.0<br/>@codecademy/gamut-labs@15.2.10-alpha.39e0b4.0<br/>@codecademy/gamut-styles@9.7.3-alpha.39e0b4.0<br/>@codecademy/gamut-tests@2.3.25-alpha.39e0b4.0<br/>@codecademy/gamut@32.8.3-alpha.39e0b4.0<br/>@codecademy/konamimojisplosion@0.2.25-alpha.39e0b4.0<br/>@codecademy/markdown-overrides@0.5.37-alpha.39e0b4.0<br/>@codecademy/styleguide@36.11.2-alpha.39e0b4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6090610a525cbd00cbb0ad59--gamut-preview.netlify.app>https://6090610a525cbd00cbb0ad59--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6090610a525cbd00cbb0ad59>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.71-alpha.3663a3.0<br/>@codecademy/gamut-labs@15.2.11-alpha.3663a3.0<br/>@codecademy/gamut@32.8.4-alpha.3663a3.0<br/>@codecademy/styleguide@36.11.3-alpha.3663a3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609152d22906321c9a5d2577--gamut-preview.netlify.app>https://609152d22906321c9a5d2577--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609152d22906321c9a5d2577>Deploy Logs</a></p>

File: packages/styleguide/stories/Molecules/Coachmark/Coachmark.examples.tsx (hunk 1 above)
Comment (codecaaron) [reply packages/styleguide/stories/Molecules/Coachmark/Coachmark.examples.tsx]: Fixed :)

**Review by JoshuaKGoldberg — APPROVED** [review]
Absolute 🔥

**Pull request merged**